ast/file_info.go

```diff
@@ -598,6 +597,46 @@ func (pos SourcePos) String() string {
 	return fmt.Sprintf("%s:%d:%d", pos.Filename, pos.Line, pos.Col)
 }
 
+func (pos SourcePos) AsSpan() SourceSpan {
+	return emptySpan{Pos: pos}
+}
+
+type emptySpan struct {
+	Pos SourcePos
+}
+
+func (e emptySpan) Start() SourcePos {
+	return e.Pos
+}
+
+func (e emptySpan) End() SourcePos {
+	return e.Pos
+}
+
+type posSpan struct {
+	StartPos SourcePos
+	EndPos   SourcePos
+}
+
+func (p posSpan) Start() SourcePos {
+	return p.StartPos
+}
+
+func (p posSpan) End() SourcePos {
+	return p.EndPos
+}
+
+func NewPosSpan(start SourcePos, end SourcePos) SourceSpan {
+	return posSpan{StartPos: start, EndPos: end}
+}
```

Comment:
```suggestion
func NewSourceSpan(start SourcePos, end SourcePos) SourceSpan {
	return posSpan{StartPos: start, EndPos: end}
}
```
Also needs Go docs.

Reply:
Oh, need Go docs for the `SourceSpan` interface, too.

Reply:
done